crates/polars-stream/src/nodes/zip.rs

```diff
@@ -176,11 +177,23 @@ impl ComputeNode for ZipNode {
                 input_head.is_broadcast == Some(false) && input_head.total_len > 0;
         }
 
-        if !self.null_extend {
-            polars_ensure!(
-                !(at_least_one_non_broadcast_done && at_least_one_non_broadcast_nonempty),
-                ShapeMismatch: "zip node received non-equal length inputs"
-            );
+        match self.extend_behavior {
+            ExtendBehavior::Broadcast => {
+                polars_ensure!(
+                    !(at_least_one_non_broadcast_done && at_least_one_non_broadcast_nonempty),
+                    ShapeMismatch: "zip node received non-equal length inputs"
+                );
+            },
+            ExtendBehavior::Raise => {
+                if let Some(first_len) = self.input_heads.first().map(|h| h.total_len) {
+                    let all_len_equal = self.input_heads.iter().all(|h| h.total_len == first_len);
+                    polars_ensure!(
+                        all_len_equal,
+                        ShapeMismatch: "zip node received non-equal length inputs"
+                    );
+                }
+            },
+            _ => {},
```

Comment:
Can you make this an explicit match for the third behavior instead of a catch-all `_`?